mypy/types.py

```diff
@@ -272,6 +273,10 @@ def __init__(self,
         self.explicit = explicit
         # Does this type come from omitted generics?
         self.from_omitted_generics = from_omitted_generics
+        # Is this type a higher-kinded type (e.g. type of call expr to NewType or NamedTuple)?
+        # We can't fully represent higher-kinded types in mypy's type system, so we treat them
+        # as 'Any' and give specific errors for each higher-kinded type (e.g. NewType).
+        self.is_higher_kinded_type = is_higher_kinded_type
```

Comment:
I'd rename this to something more general, since we don't really take advantage of the fact that these represent type constructors anywhere, and the concept of "higher-kinded type" sounds kind of complicated. My suggestion would be `special_form`, which can be applied for things other than type constructors that don't have meaningful types but that also aren't imprecisely typed.